sinabs/backend/dynapcnn/dynapcnn_layer_v2.py

```diff
@@ -0,0 +1,209 @@
+            (linear will be converted to convolutional)
+        spk: sinabs.layers.IAFSqueeze
+            Sinabs IAF layer
+        in_shape: tuple of int
+            The input shape, needed to create dynapcnn configs if the network
+            does not contain an input layer. Convention: (features, height, width)
+        pool: List of integers
+            Each integer entry represents an output (destination on chip) and
+            whether pooling should be applied (values > 1) or not (values equal
+            to 1). The number of entries determines the number of tensors the
+            layer's forward method returns.
+        discretize: bool
+            Whether to discretize parameters.
+        rescale_weights: int
+            Layer weights will be divided by this value.
+    """
+
+    def __init__(
+        self,
+        conv: nn.Conv2d,
+        spk: sl.IAFSqueeze,
+        in_shape: Tuple[int, int, int],
+        pool: List[int],
+        discretize: bool = True,
+        rescale_weights: int = 1,
+    ):
+        super().__init__()
+
+        # int conversion is done while writing the config.
+        if discretize:
+            conv, spk = discretize_conv_spike_(conv, spk, to_int=False)
+
+        self.conv               = conv
+        self.spk                = spk
+        self.in_shape           = in_shape
+        self.pool               = pool
+        self.discretize         = discretize
+        self.rescale_weights    = rescale_weights
+        self.conv_out_shape     = self._get_conv_output_shape()
+        self._pool_lyrs         = self._make_pool_layers()                                 # creates SumPool2d layers from `pool`.
+        
+
+    ####################################################### Public Methods #######################################################
+    
+    def forward(self, x):
+        """Torch forward pass.
+
+        ...
+        """
+
+        returns = []
+        
+        x = self.conv(x)
+        x = self.spk(x)
+
+        for pool in self.pool:
+            if pool == 1:
+                # no pooling is applied.
+                returns.append(x)
+            else:
+                # sum pooling of `(pool, pool)` is applied.
+                pool_out = self._pool_lyrs[pool](x)
+                returns.append(pool_out)
+
+        return tuple(returns)
+    
+    def get_neuron_shape(self) -> Tuple[int, int, int]:
+        """Return the output shape of the neuron layer.
+
+        Returns
+        -------
+        - conv_out_shape (tuple): formatted as (features, height, width).
+        """
+        # same as the convolution's output.
+        return self.conv_out_shape
+
+    def summary(self) -> dict:
+        """ Returns a summary of the convolution's/pooling's kernel sizes and the output shape of the spiking layer."""
+        # TODO I can't see pooling being used in checking memory constraints by the builder so I'm ignoring for now the fact that multiple pooling could exist.
```

Comment:
The suggestion has been incorporated, but there's something to be pointed out: with the logic up to before this change (both on the old and new implementation) the "pool" value returned by this method was always a list of length 2, whereas now this can potentially be a list with length from 0 to something larger than 2.